packages/common/react-components/src/components/DensityProvider/DensityProvider.tsx

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { createContext, PropsWithChildren } from 'react';
+
+import { Density } from '../../props';
+
+export interface DensityContextValue {
+  density?: Density;
+}
+
+export type DensityProviderProps = PropsWithChildren<{
+  density?: Density;
+}>;
+
+export const DensityContext = createContext<DensityContextValue>({ density: 'coarse' });
+
+export const DensityProvider = ({ density, children }: DensityProviderProps) => (
```

Comment:
css props might be an alternative to implement this https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#basic_usage

Reply:
I know, I've considered it. I think with something like Density, this might need to scale to affect several CSS properties, e.g. line-height, padding, margin; and I think even behavior may change, e.g. lists aren’t draggable in `fine` density.

Even with `Elevation` which would reasonably be a good target for CSS properties, Tailwind does a lot of magic with its own CSS custom properties (see [box-shadow color](https://tailwindcss.com/docs/box-shadow-color)) and trying to carry that in a custom property the design system manages proved to be unworkable.

I think for now, keeping styles and things-that-are-partly-styles in JS is our best bet.